bindings/java/src/org/sleuthkit/datamodel/SleuthkitJNI.java

```diff
@@ -231,17 +235,18 @@ public void run(String deviceId, String[] imageFilePaths) throws TskCoreExceptio
 				if (0 != tskAutoDbPointer) {
 					throw new TskCoreException("Add image process already started");
 				}
-
 				long imageHandle = openImage(imageFilePaths, false);
-				
-				synchronized (this) {					
-					tskAutoDbPointer = initAddImgNat(caseDbPointer, timezoneLongToShort(timeZone), addUnallocSpace, skipFatFsOrphans);
+				synchronized (this) {
+					if (!isCanceled) {
+						tskAutoDbPointer = initAddImgNat(caseDbPointer, timezoneLongToShort(timeZone), addUnallocSpace, skipFatFsOrphans);
+					}
 				}
 				if (0 == tskAutoDbPointer) {
 					throw new TskCoreException("initAddImgNat returned a NULL TskAutoDb pointer");
 				}
-
-				runAddImgNat(tskAutoDbPointer, deviceId, imageHandle, timeZone, imageWriterPath);
+				if (!isCanceled) {
+					runAddImgNat(tskAutoDbPointer, deviceId, imageHandle, timeZone, imageWriterPath);
+				}
```

Comment:
If we are calling stop from a cancel button in the EDT, I'm thinking it would be reasonable to give a wait cursor until the cancellation completes. Thoughts?